### ISSUE TITLE
Tool idea: convert strftime format to time.Parse format

### ISSUE THREAD

**Issue by dominikh** [dominikh/go-tools#445]
Is there any interest/use in a tool that takes a strftime format string and converts it into a format as understood by Go?

**Comment by ainar-g** [comment]
Honestly, once I've realised that the “format date” is basically `01/02 03:04:05 PM 2006 +0700`, I've never had an issue with it. If anything, I now need a tool to convert the Go date format into strftime.

**Comment by dgryski** [comment]
I'm sure there used to be a website that did this.

**Comment by dominikh** [comment]
> I'm sure there used to be a website that did this.

One with a rather offensive name, yes. To nobody's surprise, it is no longer online.

**Comment by dominikh** [comment]
Possibly nasty, possibly genius idea: detect calls such as `time.Parse("%m/%d/%Y", "...")` and provide suggested fixes (usable via gopls) that converts the format string into actual Go format. That way we don't need to provide an external tool, and get integration with editors for free. On the downside, virtually nobody would ever write a time.Parse call with a strftime format in the first place. And I'm not sure we want to encourage people to do so, either.

/cc @stamblerre

**Comment by stamblerre** [comment]
I don't think that this would necessarily encourage people since the diagnostic would pop up only if they already wrote the time with that format. I'd be supportive of any additional suggestions or quick fixes that we can add in general, and I think it would be useful to help people out with the time format as much as we can.

**Comment by fortytw2** [comment]
@dominikh IIRC there were two versions of that site, one with and one without the offensive names. 

Could you detect definitely invalid format strings? Along the lines of `01/02 03:04:05 PM 2007 +0700` or are there too many potential false positives in there for that kind of check to work, as someone could _intend_ for it to have a fixed 2007 in it

**Comment by dominikh** [comment]
@fortytw2 [SA1002](https://staticcheck.io/docs/checks#SA1002) detects definitely wrong format strings, by seeing whether `time.Parse` accepts the format or not. But like you already said, this won't and can't catch things like `2007`.

This issue is about adding a new check, which detects format strings that are valid strftime format strings, and provides an automatic fix to convert them into Go's format.
